doc/authoring_command_modules/authoring_commands.md

```diff
@@ -77,108 +95,240 @@ with self.command_group('mymod', mymod_sdk) as g:
+- `kwargs`: any supported kwarg. Commonly used kwargs include `validator`, `table_transformer`, `confirmation`, `no_wait_param` and  `transform`.
 
-==================================
+Any kwargs that are not specified will be pulled from the `command_type` kwarg, if present.
+
+***custom_command***
+
+The signature for `custom_command` is exactly the same as `command`. The only difference is that, whereas `command` uses `command_type` as the fallback for missings kwargs, `custom_command` relies on `custom_command_type`.
+
+***generic_update_command***
+
+See the section on "Suppporting Generic Update"
+
+***generic_wait_command***
+
+The generic wait command provides a templated solution for polling Azure resources until specific conditions are met.
 
-The signature of this method is 
 ```Python
-def cli_command(module_name, name, operation, client_factory=None, transform=None, table_transformer=None, confirmation=None):
+generic_wait_command(self, name, getter_name='get', getter_type=None, **kwargs)
 ```
-You will generally only specify `name`, `operation` and possibly `table_transformer`.
-  - `module_name` - The name of the module that is registering the command (e.g. `azure.cli.command_modules.vm.commands`). Typically this will be `__name__`.
-  - `name` - String uniquely naming your command and placing it within the command hierachy. It will be the string that you would type at the command line, omitting `az` (ex: access your command at `az mypackage mycommand` using a name of `mypackage mycommand`).
-  - `operation` - The handler that will be executed. Format is `<module_to_import>#<attribute_list>`
-      - For example if `operation='azure.mgmt.compute.operations.virtual_machines_operations#VirtualMachinesOperations.get'`, the CLI will import `azure.mgmt.compute.operations.virtual_machines_operations`, get the `VirtualMachinesOperations` attribute and then the `get` attribute of `VirtualMachinesOperations`.
-  - `table_transformer` (optional) - Supply a callable that takes, transforms and returns a result for table output.
-  - `confirmation` (optional) - Supply True to enable default confirmation. Alternatively, supply a callable that takes the command arguments as a dict and returning a boolean. Alternatively, supply a string for the prompt.
 
-At this point, you should be able to access your command using `az [name]` and access the built-in help with `az [name] -h/--help`. Your command will automatically be 'wired up' with the global parameters.
+- `name`: The name of the command within the command group. Commonly called 'wait'.
+- `getter_name`: The name of the method for the object getter, relative to the path specified in `operations_tmpl`.
+- `getter_type`: A `CliCommandType` object to apply to this command (optional).
+- `kwargs`: any supported kwarg.
+
+Since most wait commands rely on a simple GET call from the SDK, most of these entries simply look like:
+```Python
+   g.generic_wait_command('wait')
+```
```

Comment:
The would assume you've set up the client factory and other params previously for the group right?

Reply:
Right. It assumes an SDK-based command type is set for the group (which is usually the case)